packages/account/src/providers/provider.ts

```diff
@@ -768,49 +776,55 @@ export default class Provider {
   async getTransactionCost(
     transactionRequestLike: TransactionRequestLike,
     forwardingQuantities: CoinQuantity[] = [],
-    {
-      estimateTxDependencies = true,
-      estimatePredicates = true,
-      resourcesOwner,
-    }: TransactionCostParams = {}
-  ): Promise<TransactionCost> {
-    const transactionRequest = transactionRequestify(clone(transactionRequestLike));
+    { estimatePredicates = true, resourcesOwner }: TransactionCostParams = {}
+  ): Promise<
+    TransactionCost & {
+      estimatedInputs: TransactionRequest['inputs'];
+      estimatedOutputs: TransactionRequest['outputs'];
+    }
+  > {
+    const txRequestClone = clone(transactionRequestify(transactionRequestLike));
     const chainInfo = this.getChain();
     const { gasPriceFactor, minGasPrice, maxGasPerTx } = this.getGasConfig();
-    const gasPrice = max(transactionRequest.gasPrice, minGasPrice);
-    const isScriptTransaction = transactionRequest.type === TransactionType.Script;
+    const gasPrice = max(txRequestClone.gasPrice, minGasPrice);
+    const isScriptTransaction = txRequestClone.type === TransactionType.Script;
+
+    /**
+     * Fund with fake UTXOs to avoid not enough funds error
+     */
+    // Getting coin quantities from amounts being transferred
+    const coinOutputsQuantities = txRequestClone.getCoinOutputsQuantities();
+    // Combining coin quantities from amounts being transferred and forwarding to contracts
+    const allQuantities = mergeQuantities(coinOutputsQuantities, forwardingQuantities);
+    // Funding transaction with fake utxos
+    txRequestClone.fundWithFakeUtxos(allQuantities, resourcesOwner?.address);
```

Comment:
I had to move this above predicate estimation in order for it to work.
1. Fund with fake UTXOs,
2. Estimate predicates - if the resources owner of the fake UTXOs is a predicate, populate those resources with the predicate data so that the call to `this.estimatePredicates` actually estimates it because it internally checks for `input.predicate` being defined.